src/app/new-courses-page/course-date/course-date.component.ts

```diff
@@ -1,25 +1,93 @@
-import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+import { 
+  ChangeDetectionStrategy,
+  Component,
+  forwardRef,
+  Input,
+  OnDestroy,
+  OnInit
+} from '@angular/core';
+import { 
+  NG_VALUE_ACCESSOR,
+  ControlValueAccessor,
+  FormControl,
+  Validators,
+  AbstractControl,
+  ValidationErrors,
+  NG_VALIDATORS
+} from '@angular/forms';
+
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+export const SELECT_CONTROL_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => CourseDateComponent),
+  multi: true,
+};
+
+export const SELECT_CONTROL_VALIDATORS: any = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => CourseDateComponent),
+  multi: true,
+};
 
 @Component({
   selector: 'app-course-date',
   templateUrl: './course-date.component.html',
   styleUrls: ['./course-date.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [SELECT_CONTROL_VALUE_ACCESSOR, SELECT_CONTROL_VALIDATORS],
 })
-export class CourseDateComponent {
-  @Input()
-  get courseDate(): string { return this.date; }
-  set courseDate(date: string) {
-    if (date) {
-      this.date = date.substring(0, 10);
+export class CourseDateComponent implements ControlValueAccessor, Validators, OnInit, OnDestroy {
+  @Input() set showErrors(value: boolean) {
+    this._showErrors = value;
+    if (value) {
+      this._markFormControlAsTouched(this.formControl);
     }
   }
+  @Input() errorMsg: string;
+  
+  get showErrors() {
+    return this._showErrors;
+  }
+
+  formControl: FormControl;
+  validationMessage: string;
 
-  @Output() courseDateChange = new EventEmitter<string>();
+  private _showErrors: boolean = false;
+  private _destroy$: Subject<void> = new Subject();
+  private _onTouched: () => void = () => {};
+
+  ngOnInit(): void {
+    this.formControl = new FormControl('', [Validators.required]);
+  }
+
+  ngOnDestroy(): void {
+    this._destroy$.next();
+    this._destroy$.complete();
+  }
 
-  private date: string;
+  writeValue(val): void {
+    if (!val) {
```

Comment:
what if you want to reset the control value by passing null? You need to handle that case as well.